model-optimizer/extensions/middle/StridedSliceReplacer.py

```diff
@@ -0,0 +1,64 @@
+# Copyright (C) 2018-2021 Intel Corporation
+# SPDX-License-Identifier: Apache-2.0
+
+import numpy as np
+
+from extensions.middle.InsertLayoutPropagationTransposes import InsertLayoutPropagationTranspose
+from extensions.middle.StridedSliceNormalizer import StridedSliceNormalizer
+from mo.front.common.partial_infer.utils import int64_array
+from mo.front.tf.graph_utils import create_op_node_with_second_input
+from mo.graph.graph import Graph, rename_nodes, Node
+from mo.middle.replacement import MiddleReplacementPattern
+from mo.ops.squeeze import Squeeze
+from mo.ops.unsqueeze import Unsqueeze
+
+
+def replace_strided_slice(node: Node, mask: np.ndarray, op: callable):
+    node_name = node.soft_get('name', node.id)
+    axes = np.where(mask == 1)[0]
+    new_node = create_op_node_with_second_input(node.graph, op, int64_array(axes))
+    node.in_port(0).get_connection().set_destination(new_node.in_port(0))
+    node.out_port(0).get_connection().set_source(new_node.out_port(0))
+
+    rename_nodes([(node, node_name + '/ShouldBeDeleted'), (new_node, node_name)])
+    node.graph.remove_node(node.id)
```

Comment:
please use `graph.remove_nodes_from([node.id])` for deleting a node. One you use is from old Graph API.